dependency/src/main/java/de/dagere/peass/dependency/DependencyManager.java

```diff
@@ -243,29 +243,76 @@ public void updateDependenciesOnce(final TestCase testcase, final File parent, f
 
       LOG.debug("Test: {} ", testcase);
       LOG.debug("Dependencies: {}", allCalledClasses.size());
-      dependencies.setDependencies(testcase, allCalledClasses);
+      
+      boolean testExcluded = false;
+
+      if (forbiddenMethodCalled(allCalledClasses)) {
+         testExcluded = true;
+         fakeConfig.getExecutionConfig().getExcludes().add(testcase.getExecutable());
+      }
+      
+      if (getTraceSize(kiekerResultFolders) > fakeConfig.getKiekerConfig().getTraceSizeInMb()) {
+         testExcluded = true;
+      }
+
+      if (!testExcluded) {
+         dependencies.setDependencies(testcase, allCalledClasses);
+      }
    }
 
-   private Map<ChangedEntity, Set<String>> getCalledMethods(final ModuleClassMapping mapping, final File[] kiekerResultFolders) {
-      final Map<ChangedEntity, Set<String>> allCalledClasses = new LinkedHashMap<ChangedEntity, Set<String>>();
+   private boolean forbiddenMethodCalled(final Map<ChangedEntity, Set<String>> allCalledClasses) {
+      List<String> forbiddenMethods = fakeConfig.getExecutionConfig().getForbiddenMethods();
+
+      for (String forbiddenMethod : forbiddenMethods) {
+         if (traceContainsMethod(allCalledClasses, forbiddenMethod)) {
+            return true;
+         }
+      }
+
+      return false;
+   }
+
+   private boolean traceContainsMethod(final Map<ChangedEntity, Set<String>> allCalledClasses, final String methodToFind) {
+      String[] classAndMethod = methodToFind.split("#");
+
+      for (Entry<ChangedEntity, Set<String>> calledClass : allCalledClasses.entrySet()) {
+         if (calledClass.getKey().getClazz().equals(classAndMethod[0])) {
+            Set<String> calledMethods = calledClass.getValue();
+            if (calledMethods.contains(classAndMethod[1])) {
+               return true;
+            }
+         }
+      }
+
+      return false;
+   }
+
+   private long getTraceSize(final File[] kiekerResultFolders) {
+      long overallSizeInMB = 0;
       for (File kiekerResultFolder : kiekerResultFolders) {
          final long size = FileUtils.sizeOfDirectory(kiekerResultFolder);
          final long sizeInMB = size / (1024 * 1024);
+         overallSizeInMB += sizeInMB;
 
          LOG.debug("Size: {} Folder: {}", sizeInMB, kiekerResultFolder);
-         if (sizeInMB > fakeConfig.getKiekerConfig().getTraceSizeInMb()) {
```

Comment:
This part of the diffs is confusing. Only the first case of the if-else block was deleted. The new version is shown in the last method of the diffs.